zulip_bots/zulip_bots/lib.py

```diff
@@ -102,7 +102,74 @@ def __init__(self, name: str, email: str) -> None:
         self.email = email
         self.mention = '@**' + name + '**'
 
-class ExternalBotHandler:
+class BotHandler(Protocol):
+
+    email: str
+    full_name: str
+    storage: Any
+
+    def reset_transcript(self) -> None:
+        pass
+
+    def identity(self) -> BotIdentity:
+        pass
+
+    def react(
+        self,
+        message: Dict[str, Any],
+        emoji_name: str
+    ) -> Dict[str, Any]:
+        pass
+
+    def send_message(
+        self,
+        message: (Dict[str, Any])
+    ) -> Dict[str, Any]:
+        pass
+
+    def send_reply(
+        self,
+        message: Dict[str, Any],
+        response: str,
+        widget_content: Optional[str] = None
+    ) -> Dict[str, Any]:
+        pass
```

Comment:
Should this be `...` ?

Reply:
I am not sure about what you are saying! Do you mean I should use *args or **kwargs for this because of the EmbeddedBotHandler in Zulip?

Reply:
I am not sure of what is `...`!! Do you mean these should be in 1 line?

Reply:
https://mypy.readthedocs.io/en/stable/protocols.html#simple-user-defined-protocols

Reply:
Ahhh, ok I understand now. Probably missed this part while going through the library.

Reply:
Thanks, I resolved it.